src/client/service-worker/events/utility.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  match, put, remove, getMetaData, setMetaData, createCacheName,
+} from '@americanexpress/one-service-worker';
+
+const localeRegExp = /(([a-z]{2,3})(?:-)?([a-zA-Z]{1,})?)\/[^/]*\.json/;
+const legacyRegExp = /legacy\.browser|\/legacy\//;
+
+export function createResourceMetaData(event, resourceInfo, revision) {
```

Comment:
I don't see any unit tests for these functions, this is really important especially when introducing complex regexs

Reply:
The goal was to test the implementation, not the utility functions per se but that the system works and is covered. What do you think makes the RegExps complex?

Reply:
I've broken up the `regexp` and simplified it to one main capture group if necessary for matching (efd5586)

Reply:
The other thing is we need to have some BCP 47 shaped regexp which is why it may be more involved.

Reply:
I am not able to look at that regex and instantly understand all the patterns which would fail and match, you might be a regex wizard but not everyone who works on or will work on will be. These functions are perfect examples of where unit tests are necessary. The unit tests not only protect from regression but they help convey and show the intended api for the function. You need to test the implementation as a whole and at the level of each function.